opus/application/static_media/js/browse.js

```diff
@@ -548,34 +521,94 @@ var o_browse = {
                     Left: 37
                     Space: 32 */
 
+                let tab = opus.getViewTab();
+                let viewNamespace = opus.getViewNamespace();
                 let detailOpusId;
+                let offset = 0;
+                let obsNum = $("#galleryViewContents .op-obs-direction a").data("obs");
+                let lastCachedObsNum = $(`${tab} .op-thumbnail-container`).last().data("obs");
                 // the || is for cross-browser support; firefox does not support keyCode
                 switch (e.which || e.keyCode) {
                     case 32:  // spacebar
-                        if (opus.metadataDetailOpusId !== "") {
+                        if (opus.metadataDetailOpusId !== "" && opus.metadataDetailOpusId !== undefined && !$("#galleryViewContents").hasClass("op-disabled")) {
                             o_browse.undoRangeSelect();
                             o_cart.toggleInCart(opus.metadataDetailOpusId);
                         }
                         break;
                     case 39:  // next
-                        detailOpusId = $("#galleryView").find(".op-next").data("id");
-                        o_browse.removeEditMetadataDetails();
-                        o_browse.loadPageIfNeeded("next", detailOpusId);
+                        obsNum++;
+                        // if this new observation is in the last row of cached observations, don't allow the user to arrow there
```

Comment:
Isn't this comment wrong? I thought we only disallowed the last row if a load was in progress. In fact I don't understand this at all - I can move the binocs to the last row, but this code would seem to prevent that. What am I missing?

Reply:
Also, if you go to the table view, make enough columns that there's a horizontal scrollbar, go to the very last observation, and hit the right arrow, the scrollbar will move rather than just ignoring the arrow like we should because it's now captured by the slide.